python/ray/runtime_context.py

```diff
@@ -160,6 +161,25 @@ def should_capture_child_tasks_in_placement_group(self):
         """
         return self.worker.should_capture_child_tasks_in_placement_group
 
+    def get_assigned_resources(self):
+        """Get the assigned resources to this worker.
+
+        By default for tasks, this will return {"CPU": 1}.
+        By default for actors, this will return {}.
+
+        Returns:
+            A dictionary mapping the name of a resource to a float, where
+            the float represents the amount of that resource reserved
+            for this worker.
+        """
+        self.worker.check_connected()
+        resource_id_map = self.worker.core_worker.resource_ids()
```

Comment:
What happens if we invoke this API inside a driver? Should we raise an explicit exception?

Reply:
Added an assertion for worker mode.